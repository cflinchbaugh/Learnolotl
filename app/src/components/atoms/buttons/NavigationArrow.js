import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    background-color: ${props => props.theme.secondary};
    color: ${props => props.theme.secondaryColor};
    transition: all 0.35s; 
    display: inline-block;
    border-radius: 15px;
    padding: 8px 20px;
    box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.25);
    user-select: none;

    &:hover {
        cursor: pointer;
        background-color: ${props => props.theme.secondaryLight};
        color: ${props => props.theme.secondaryLightColor}
        box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.48);
    }
`

function Button (props) {
    let arrow = props.type === 'next' ? '=>' : '<=';

    return (
        <StyleWrapper 
            className="navigation-button"
            onClick={ props.onClickFunction } >
                {arrow}
        </StyleWrapper>
    )
}

export default Button