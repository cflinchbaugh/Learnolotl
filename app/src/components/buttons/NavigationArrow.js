import React from 'react';
import styled from 'styled-components';

const StyleWrapper = styled.div`
    color: orange;
    border: solid 1px orange;
    transition: all 0.35s; 
    display: inline-block;
    border-radius: 15px;
    padding: 8px 20px;
    box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.25);
    user-select: none;

    &:hover {
        cursor: pointer;
        background-color: #566184;
        color: ${props => props.theme.secondaryLightColor}
        box-shadow: 1px 8px 19px 3px rgba(61,55,61,0.48);
    }
`

function NavigationArrow (props) {
    let arrow = props.type === 'next' ? '=>' : '<=',
        navigationArrowData ={
            className: 'navigation-button',
            onClick: props.onClickFunction
        }

    return (
        <StyleWrapper {...navigationArrowData} >
            {arrow}
        </StyleWrapper>
    )
}

export default NavigationArrow