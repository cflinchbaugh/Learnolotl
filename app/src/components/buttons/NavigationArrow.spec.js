import React from 'react';
import {shallow} from 'enzyme';
import NavigationArrow from './NavigationArrow';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('NavigationArrow', () => {
    it('renders and matches snapshot', () => {
        const renderer = new ShallowRenderer();

        const tree = renderer.render(
            <NavigationArrow />
        );

        expect(tree).toMatchSnapshot();
    });

    it('should call handleBuildLandingViewNext when the NavigationArrow is clicked', () => {
        const onClickFunction = jest.fn();

        const wrapper = shallow(<NavigationArrow onClickFunction={onClickFunction}/>);

            wrapper.find('.navigation-button').simulate('click');

            expect(onClickFunction).toHaveBeenCalled();
    });
});