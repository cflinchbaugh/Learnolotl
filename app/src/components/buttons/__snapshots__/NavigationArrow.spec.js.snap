// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationArrow renders and matches snapshot 1`] = `
<styled.div
  className="navigation-button-wrapper"
>
  <button
    className="navigation-button"
    onClick={undefined}
  >
    &lt;=
  </button>
</styled.div>
`;
